src/main/java/ch/heigvd/amt/mvcProject/application/question/QuestionFacade.java

```diff
@@ -153,6 +149,26 @@ public QuestionsDTO getQuestions(QuestionQuery query) throws QuestionFailedExcep
         return questionFound;
     }
 
+    /**
+     * Get a question with a query that contains an UserID
+     * @param query query
+     * @return questions found
+     */
+    public QuestionsDTO getQuestionsByUserId(QuestionQuery query) {
+        Collection<Question> userQuestions = questionRepository.findByUserId(query.userId);
```

Comment:
ça me gêne pas qu'en soit on sépare getQuestions en plusieurs méthodes, mais le check ne parte pas, si query.userId est null, ça va crash ce qui est pas bon

aussi, alors, c'est soit on met tout dans getQuestions en multipliant les if, else if, ce qui est en soi la méthode la plus logique puisque dans le cas ici, nous avons un problème si on veut rechercher toutes les questions de l'utilisateur mais qui contiennent dans leur titre 'C++', mais on va pas le faire ici, car "I don't have the time for that". Ici on peut traiter un seul champ à la fois -> le titre, les tags, le contenu, etc. mais si on devait vraiment bien faire la facade (ce qui devrait être le cas), il est nécessaire de bien prendre en compte qu'on devrait pouvoir réaliser une requête de type :

Je voudrais toutes les questions qui ont dans leur titre 'template', qui ont les tags 'C++', 'Java' et qui ont été posées par l'utilisateur 'Paul'